### PR TITLE
[@types/react-adaptive-hooks] Add initial typings

### PR DESCRIPTION
**Pull request by stramel**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Moved to DefinitelyTyped from the official repo. https://github.com/GoogleChromeLabs/react-adaptive-hooks/pull/21

### CODE REVIEW

File: types/react-adaptive-hooks/media-capabilities.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    smooth: boolean;
+    powerEfficient: boolean;
+}
+
+export function useMediaCapabilities<T = Partial<MediaConfigurationInfo>>(
+    mediaConfig: MediaConfiguration,
+    initialMediaCapabilities: T,
+): {
+    mediaCapabilities:
+        | ({ supported: boolean; hasMediaConfig: boolean } & T) | ({ hasMediaConfig?: undefined} & MediaConfigurationInfo);
```
Comment (43081j) [inline types/react-adaptive-hooks/media-capabilities.d.ts]: this seems weird. is the aim that if `hasMediaConfig` is true, you have media configuration info? you should be able to do similar to your `unsupported` types here if so

File: types/react-adaptive-hooks/memory.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export interface MemoryStatus {
+    deviceMemory: number;
+    totalJSHeapSize: number | null;
```
Comment (43081j) [inline types/react-adaptive-hooks/memory.d.ts]: can these just be optional? `totalJSHeapSize?: number;`

Comment (43081j) [reply types/react-adaptive-hooks/memory.d.ts]: are they likely to ever be `null` explicitly, is what i mean

Comment (stramel) [reply types/react-adaptive-hooks/memory.d.ts]: They are explicitly set to `null` if not defined.

File: types/react-adaptive-hooks/media-capabilities.d.ts (hunk 1 above)
Comment (stramel) [reply types/react-adaptive-hooks/media-capabilities.d.ts]: This whole hook is pretty rough. The code isn't even correct yet but has been released. 👎 Not entirely sure what the point of it is but it's in the code.

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Comment by stramel** [comment]
@uniqueiniquity not my first contribution but thanks for the merge 🙂

**Pull request merged**